lib/internal/Magento/Framework/Message/Manager.php

```diff
@@ -79,16 +81,16 @@ public function __construct(
         Event\ManagerInterface $eventManager,
         LoggerInterface $logger,
         $defaultGroup = self::DEFAULT_GROUP,
-        ExceptionMessageFactoryInterface $exceptionMessageFactory = null
+        ExceptionMessageFactoryInterface $exceptionMessageFactory = null,
+        ExceptionMessageLookupFactory $exceptionMessageLookupFactory = null
     ) {
         $this->session = $session;
         $this->messageFactory = $messageFactory;
         $this->messagesFactory = $messagesFactory;
         $this->eventManager = $eventManager;
         $this->logger = $logger;
         $this->defaultGroup = $defaultGroup;
-        $this->exceptionMessageFactory = $exceptionMessageFactory ?: ObjectManager::getInstance()
-            ->get(ExceptionMessageLookupFactory::class);
+        $this->exceptionMessageFactory = $exceptionMessageFactory ?: $exceptionMessageLookupFactory;
```

Comment:
To make this change backward compatible, please use ObjectManager to initialize `$exceptionMessageLookupFactory`